rice-middleware/impl/src/main/java/org/kuali/rice/kim/service/impl/UiDocumentServiceImpl.java

```diff
@@ -2618,6 +2619,7 @@ protected void convertCheckboxAttributeData(RoleMemberAttributeDataBo roleMember
 			for(RoleDocumentDelegation roleDocumentDelegation: identityManagementRoleDocument.getDelegations()){
 				newKimDelegation = new DelegateTypeBo();
 				KimCommonUtilsInternal.copyProperties(newKimDelegation, roleDocumentDelegation);
+                newKimDelegation = KRADServiceLocator.getDataObjectService().copyInstance(newKimDelegation, CopyOption.RESET_VERSION_NUMBER,CopyOption.RESET_OBJECT_ID);

```

Comment:
Oh, I see.  copyInstance copies only the same object.  With copyProperties we copy the properties from RoleDocumentDelegation object to a DelegateTypeBo object.

The copyInstance in this case just copies itself while setting the version number and object id to null.  Which of course makes me wonder why we don't just set them to null. 

    newKimDelegation.setVersionNumber(null);
    newKimDelegation.setObjectId(null);